object_manager_test.go

```diff
@@ -349,6 +351,151 @@ var _ = Describe("Object Manager", func() {
+
+		anFakeConnector.createObjectObj.(*IPv6Network).Ea = objMgr.getBasicEA(true)
+		anFakeConnector.createObjectObj.(*IPv6Network).Ea["Network Name"] = networkName
+
+		var actualNetwork *IPv6Network
+		var err error
+		It("should pass expected IPv6 Network Object to CreateObject", func() {
+			actualNetwork, err = objMgr.AllocateIPv6Network(netviewName, cidr, prefixLen, networkName)
+		})
+		It("should return expected IPv6 Network Object", func() {
+			Expect(actualNetwork).To(Equal(anFakeConnector.resultObject))
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("Get IPv6 Network", func() {
+		cmpType := "Docker"
+		tenantID := "01234567890abcdef01234567890abcdef"
+		netviewName := "default_view"
+		cidr := "2001:db8:abcd:0012::0/64"
+		networkName := "private-net"
+		ea := EA{"Network Name": networkName}
+		fakeRefReturn := fmt.Sprintf("ipv6network/ZG5zLm5ldHdvcmskODkuMC4wLjAvMjQvMjU:%s/%s", cidr, netviewName)
+		nwFakeConnector := &fakeConnector{
+			getObjectObj: NewIPv6Network(netviewName, cidr, ea),
+			getObjectRef: "",
+			resultObject: []IPv6Network{*NewIPv6Network(netviewName, cidr, nil)},
+		}
+
+		nwFakeConnector.resultObject.([]IPv6Network)[0].Ref = fakeRefReturn
+		nwFakeConnector.getObjectObj.(*IPv6Network).eaSearch = EASearch(ea)
+		nwFakeConnector.resultObject.([]IPv6Network)[0].eaSearch = EASearch(ea)
+
+		objMgr := NewObjectManager(nwFakeConnector, cmpType, tenantID)
+
+		var actualNetwork *IPv6Network
+		var err error
+		It("should pass expected IPv6 Network Object to GetObject", func() {
+			actualNetwork, err = objMgr.GetIPv6Network(netviewName, cidr, ea)
+		})
+		It("should return expected IPv6 Network Object", func() {
+			Expect(*actualNetwork).To(Equal(nwFakeConnector.resultObject.([]IPv6Network)[0]))
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("Get IPv6 Network with Reference", func() {
+		cmpType := "Docker"
+		tenantID := "01234567890abcdef01234567890abcdef"
+		cidr := "2001:db8:abcd:0012::0/64"
+		netviewName := "default_view"
+		getRef := fmt.Sprintf("ipv6network/ZG5zLm5ldHdvcmskODkuMC4wLjAvMjQvMjU:%s/%s", cidr, netviewName)
+		fakeRefReturn := getRef
+		nwFakeConnector := &fakeConnector{
+			getObjectObj:  NewIPv6Network("", "", nil),
+			getObjectRef:  getRef,
+			resultObject:  []IPv6Network{*NewIPv6Network("", "", nil)},
+			fakeRefReturn: fakeRefReturn,
+		}
+
+		objMgr := NewObjectManager(nwFakeConnector, cmpType, tenantID)
+
+		var actualRef *IPv6Network
+		var err error
+		It("should pass expected IPv6 Network Ref to getObject", func() {
+			actualRef, err = objMgr.GetIPv6NetworkWithRef(fakeRefReturn)
+		})
+		It("should return expected IPv6 Network record Ref", func() {
+			Expect(*actualRef).To(Equal(nwFakeConnector.resultObject.([]IPv6Network)[0]))
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("Delete IPv6 Network", func() {
+		cmpType := "Docker"
+		tenantID := "01234567890abcdef01234567890abcdef"
+		netviewName := "default_view"
+		cidr := "2001%3Adb8%3Aabcd%3A0012%3A%3A0/64"
```

Comment:
I think, this CIDR is not a valid one. Why '%3A' character sequence is used instead of ':'?